pootle/apps/pootle_format/utils.py

```diff
@@ -63,3 +76,99 @@ def valid_extensions(self):
             OrderedDict.fromkeys(
                 self.filetype_extensions
                 + self.template_extensions))
+
+    def add_filetype(self, filetype):
+        """Adds a filetype to a Project
+        """
+        if filetype not in self.filetypes:
+            log.info(
+                "Adding filetype '%s' to project '%s'",
+                filetype, self.project)
+            self.project.filetypes.add(filetype)
+
+    def set_store_filetype(self, store, filetype):
+        """Sets a Store to given filetype
+
+        If `from_filetype` is given, only updates if current filetype matches
+        """
+        if filetype not in self.filetypes:
+            raise self.unrecognized_filetype(filetype)
+        log.info(
+            "Setting filetype '%s' to store '%s'",
+            filetype, store.pootle_path)
+        store.filetype = filetype
+        # update the extension if required
+        extension = (
+            store.is_template
+            and str(filetype.template_extension)
+            or str(filetype.extension))
+        root_name = os.path.splitext(store.name)[0]
+        new_name = (
+            root_name.endswith(".%s" % extension)
+            and root_name
+            or ("%s.%s"
+                % (root_name, extension)))
```

Comment:
IMHO this is more readable:

```python

            if root_name.endswith(".%s" % extension):
                new_name = root_name
            else:
                new_name = "%s.%s" % (root_name, extension)
```

Using this `x and y or z` is better if it is shorter than a regular `if x: y else: z` (usually when the first is a one-liner), but otherwise you are just losing readability. And in this case is longer too.